flink-table-store-core/src/main/java/org/apache/flink/table/store/file/catalog/FileSystemCatalog.java

```diff
@@ -0,0 +1,298 @@
+        }
+
+        uncheck(() -> fs.delete(path, true));
+    }
+
+    @Override
+    public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ignoreIfExists)
+            throws TableAlreadyExistException, DatabaseNotExistException, CatalogException {
+        if (!databaseExists(tablePath.getDatabaseName())) {
+            throw new DatabaseNotExistException(getName(), tablePath.getDatabaseName());
+        }
+
+        Path path = tablePath(tablePath);
+        if (tableExists(path)) {
+            if (ignoreIfExists) {
+                return;
+            }
+
+            throw new TableAlreadyExistException(getName(), tablePath);
+        }
+
+        commitTableChange(path, table);
+    }
+
+    @Override
+    public void alterTable(
+            ObjectPath tablePath, CatalogBaseTable newTable, boolean ignoreIfNotExists)
+            throws TableNotExistException, CatalogException {
+        Path path = tablePath(tablePath);
+        if (!tableExists(path)) {
+            if (ignoreIfNotExists) {
+                return;
+            }
+
+            throw new TableNotExistException(getName(), tablePath);
+        }
+
+        commitTableChange(path, newTable);
+    }
+
+    private static <T> T uncheck(Callable<T> callable) {
+        try {
+            return callable.call();
+        } catch (Exception e) {
+            if (e instanceof RuntimeException) {
+                throw (RuntimeException) e;
+            }
+            throw new CatalogException(e);
+        }
+    }
+
+    private static void uncheck(RunnableWithException runnable) {
+        FileSystemCatalog.uncheck(
+                () -> {
+                    runnable.run();
+                    return null;
+                });
+    }
+
+    private static ClassLoader classLoader() {
+        return FileSystemCatalog.class.getClassLoader();
+    }
+
+    private static boolean isDatabase(Path path) {
+        return path.getName().endsWith(DB_SUFFIX);
+    }
+
+    private static String database(Path path) {
+        String name = path.getName();
+        return name.substring(0, name.length() - DB_SUFFIX.length());
+    }
+
+    private Path databasePath(String database) {
+        return new Path(root, database + DB_SUFFIX);
+    }
+
+    private Path tablePath(ObjectPath objectPath) {
+        return new Path(databasePath(objectPath.getDatabaseName()), objectPath.getObjectName());
+    }
+
+    private void commitTableChange(Path tablePath, CatalogBaseTable baseTable) {
+        if (!(baseTable instanceof ResolvedCatalogTable)) {
+            throw new UnsupportedOperationException(
+                    "Only support ResolvedCatalogTable, but is: " + baseTable.getClass());
+        }
+        ResolvedCatalogTable table = (ResolvedCatalogTable) baseTable;
+
+        // remove table path
+        Map<String, String> options = table.getOptions();
+        String specific = options.remove(PATH.key());
```

Comment:
I have a question here.

For catalog created table, `Schema` does not record table path, but for managed table, table path are kept, which is inconsistent. Is it by design?

Reply:
Yes, 
- for catalog table, it has a warehouse context, it can re-produce the table path when get catalog table.
- but for managed, it has no warehouse context, which means we must store table path or warehouse path for it.